src/server/db_slice.cc

```diff
@@ -468,6 +468,7 @@ OpResult<DbSlice::PrimeItAndExp> DbSlice::FindInternal(const Context& cntx, std:
       db.prime.CVCUponBump(change_cb_.back().first, res.it, bump_cb);
     }
 
+    block_counter_.Wait();  // We must not change the bucket's internal order during serialization
```

Comment:
not related to this PR but `block_counter_` is not a good name because it does not say much about the semantics of the variable.